src/transform/desugar_destructuring.rs

```diff
@@ -54,23 +55,26 @@ pub fn transform_one(rt: RichTerm) -> RichTerm {
 /// down traversal. This means that the return value is a normal `Term::Fun` but it can contain
 /// `Term::FunPattern` and `Term::LetPattern` inside.
 pub fn desugar_fun(rt: RichTerm) -> RichTerm {
-    match_sharedterm! {rt.term, with {
-        Term::FunPattern(x, pat, t_) if !pat.is_empty() => {
-            let x = x.unwrap_or_else(Ident::fresh);
-            let t_pos = t_.pos;
-            RichTerm::new(
-                Term::Fun(
-                    x,
-                    RichTerm::new(Term::LetPattern(None, pat, Term::Var(x).into(), t_), t_pos /* TODO: should we use rt.pos? */),
-                ),
-                rt.pos,
-            )
-        },
-        Term::FunPattern(Some(x), Destruct::Empty, t_) => RichTerm::new(Term::Fun(x, t_), rt.pos),
-        t@Term::FunPattern(..) => panic!(
-            "A function can not have empty pattern without name in {t:?}"
-        ),
-    } else rt
+    match_sharedterm! { rt.term,
+        with {
+            Term::FunPattern(x, pat, t_) => {
+                let x = x.unwrap_or_else(Ident::fresh);
+                let t_pos = t_.pos;
+                RichTerm::new(
+                    Term::Fun(
+                        x,
+                        RichTerm::new(Term::LetPattern(None, pat, Term::Var(x).into(), t_), t_pos /* TODO: should we use rt.pos? */),
+                    ),
+                    rt.pos,
+                )
+            },
+            t@Term::FunPattern(..) => panic!(
+                "A function can not have empty pattern without name in {t:?}"
+            ),
```

Comment:
Isn't this dead code now?
```suggestion
```

Reply:
Ah, yeah, good point - I guess the compiler doesn't notice because it's inside the `match_sharedterm!` macro. I would have thought dead code analysis happens after macro expansion, but I guess not... :thinking:

Reply:
Yeah, I wondered why it wasn't flagged as well.